provider.go

```diff
@@ -309,44 +308,73 @@ func (p *Provider) putMedia(baseTimecode chan uint64, ch chan ebml.Block, chTag
 		},
 	}
 
-	r, wOut := io.Pipe()
-	w := io.Writer(wOut)
+	r, wOutRaw := io.Pipe()
+	wOutBuf := bufio.NewWriter(wOutRaw)
+	writeErr := func() error { return nil }
+	var w io.Writer
 	var backup *bytes.Buffer
+
 	if opts.retryCount > 0 {
+		// Ignore error when http request body is closed.
+		// Continue marshalling whole fragment and retry sending later.
+		noErrWriter := &ignoreErrWriter{Writer: wOutBuf}
+		w = io.Writer(noErrWriter)
```

Comment:
This line's `w`  will be overridden soon by L328 `w = io.MultiWriter(wOutBuf, backup)`, so that the `noErrWriter` seems not to be used anywhere. Should `noErrWriter` be passed to `io.MultiWriter` instead of  `wOutBuf` on L328?

```go
		w = io.MultiWriter(noErrWriter, backup)
```

Reply:
Fixed and adding a test case for http disconnection.